scenes/ia_hard.go

```diff
@@ -151,7 +164,7 @@ func (s *iaHard) Run(win *pixelgl.Window) Scene {
 		bestWeights = best.Neural().Weights()
 		bestPoints = best.Bird().Points
 
-		err := best.Neural().Dump(datafile)
+		err := best.Neural().Dump(s.generation+1, datafile)
```

Comment:
Acho que é melhor salvar a geração atual mesmo no arquivo, e quando for carregar dai incrementar. Porque dai fica registrada a real geração que "exportou" esses dados.